app/src/main/java/com/example/sharingang/ui/fragments/DetailedItemFragment.kt

```diff
@@ -198,80 +194,79 @@ class DetailedItemFragment : Fragment() {
-            val selectedOption: Int = binding.radioGroup.checkedRadioButtonId
-            if (selectedOption != -1) {
-                val rating = when (selectedOption) {
-                    binding.radioButton1.id -> 1
-                    binding.radioButton2.id -> 2
-                    binding.radioButton3.id -> 3
-                    binding.radioButton4.id -> 4
-                    binding.radioButton5.id -> 5
-                    else -> 0
-                }
-                viewModel.updateUserRating(item.userId, rating)
-                itemViewModel.rateItem(item)
+            val rating = when (binding.radioGroup.checkedRadioButtonId) {
+                binding.radioButton1.id -> 1
+                binding.radioButton2.id -> 2
+                binding.radioButton3.id -> 3
+                binding.radioButton4.id -> 4
+                binding.radioButton5.id -> 5
+                else -> 0
             }
+            viewModel.updateUserRating(item.userId, rating)
+            itemViewModel.updateReview(item, currentUserProvider.getCurrentUserId(), false)
         }
     }
 
-
-    private fun getButtonText(contains: Boolean): String {
-        return if (contains) getString(R.string.remove_wishlist)
-        else getString(R.string.add_wishlist)
-    }
-
     private fun updateWishlist() {
         viewModel.modifyWishList(args.item)
     }
 
-    /**
-     * Start the buying process
-     *
-     * On success, the item is marked as sold and the rating form is displayed
-     */
     private fun buyItem() {
-        binding.buyButton.isEnabled = false
-        lifecycleScope.launch {
-            val status = paymentProvider.requestPayment(args.item)
-            if (status) {
-                itemViewModel.sellItem(args.item)
-                binding.buyButton.visibility = View.GONE
-                initRating(args.item.copy(sold = true))
-            } else {
-                binding.buyButton.isEnabled = true
+        val quantity: Int = binding.quantity?.toIntOrNull() ?: 1
+        if (quantity > args.item.quantity) {
+            Toast.makeText(context, "Not enough available", Toast.LENGTH_SHORT).show()
+            binding.buyButton.isEnabled = true
+        } else {
+            lifecycleScope.launch {
+                if (paymentProvider.requestPayment(args.item, quantity)) {
+                    itemViewModel.sellItem(args.item)
+                    val newQuantity = args.item.quantity - quantity
+                    binding.sellerVisibility = if (newQuantity == 0) View.GONE else View.VISIBLE
+                    binding.buyButton.isEnabled = newQuantity != 0
+                    updateBoughtItem(args.item, newQuantity)
+                } else {
+                    binding.buyButton.isEnabled = true
+                }
             }
         }
     }
 
+    private fun updateBoughtItem(item: Item, quantity: Int) {
+        binding.detailedItemQuantity.text = String.format("Quantity: %s", quantity.toString())
```

Comment:
Doesn't it work if you do binding.item = itemToUpdate, that way the text field for the quantity takes the new quantity of the item directly, doesn't it?

Reply:
yup thanks that works and is a bit cleaner.